services/src/test/java/org/exoplatform/addons/gamification/service/configuration/DomainServiceTest.java

```diff
@@ -67,10 +68,17 @@ public void testGetAllDomains() {
   }
 
   @Test
-  public void testFindDomainByTitle() {
-    assertNull(domainService.findDomainByTitle(GAMIFICATION_DOMAIN));
-    newDomain();
-    assertNotNull(domainService.findDomainByTitle(GAMIFICATION_DOMAIN));
+  public void testFindEnabledDomainByTitle() {
+    assertNull(domainService.findEnabledDomainByTitle(GAMIFICATION_DOMAIN));
+    DomainDTO domainDTO = newDomainDTO();
+    assertNotNull(domainService.findEnabledDomainByTitle(GAMIFICATION_DOMAIN));
+    domainDTO.setEnabled(false);
+    try {
+      domainService.updateDomain(domainDTO);
+    } catch (ObjectNotFoundException e) {
+    //
```

Comment:
should it fail when it catches an exception ?

Reply:
Could you explain when we use getDomainByTitle and when we use findEnabledDomainBytitle ?

Reply:
in our case for the achievement table after deactivating a domain the program ID field is empty, we need to display its program even if it is disabled we use in this case getDomainByTitle also for the zoneRegistry
 in other cases we use findEnabledDomainBytitle